src/main/java/io/fusionauth/samlv2/service/DefaultSAMLv2Service.java

```diff
@@ -271,30 +312,33 @@ public String buildAuthnResponse(AuthenticationResponse response, boolean sign,
 
     Document document = marshallToDocument(PROTOCOL_OBJECT_FACTORY.createResponse(jaxbResponse), ResponseType.class);
 
-    // Bail now if we aren't signing the response.
-    if (!sign) {
+    // Sign the Assertion if requested
```

Comment:
This had to be refactored to support a mix of signing and encryption.
* If the `Assertion` is signed, it should be signed _before_ encrypting
* If the `Response` is signed, it should be signed _after_ encrypting the assertion